File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>
int main(void)
{
    string text = get_string("Text: ");
    int letters = 0,  words = 1,  sentences = 0;


   //iterate over the string to get words letters and sentences y
    for (int i = 0; i < strlen(text); i++)
    {
       if (isalpha(text[i]))
       {
           letters++;
       }
       else if (isspace(text[i]))
       {
           words++;
       }
       else if (text[i] == '!' || text[i] == '.' || text[i] == '?')
       {
           sentences++;
       }
    }
    // to retrive grade index

    float g = 0.0588 * (100 * (float) letters / (float) words) - 0.296 * (100 * (float) sentences / (float) words) - 15.8;
    if (g < 16 && g >= 0)
    {
        printf("Grade %i\n", (int) round(g));
    }
    else if (g >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }

}

File: pset6/readability/readability.py
from cs50 import get_string


text = get_string("Text: ")
text_length = len(text)

word_count = 1
letter_count = 0  
sentence_count = 0

### Number of letters ###
for i in range(text_length):
    if(text[i].isalpha()):
       letter_count = letter_count + 1
    
    ## Get number of sentences when terminating characters ., ?, ! ###   
    if(text[i] == '.' or text[i] == '?' or text[i] == '!'):
        sentence_count +=1
    
    
    ## Number of word by detecting spaces 
    if (text[i].isspace()):
        word_count +=1
     ## calculate index   
calculation = (0.0588 * letter_count / word_count * 100) - (0.296 * sentence_count / word_count * 100) - 15.8
    
index = round(calculation)
if index<1:
    print("Before Grade 1")
elif index>16:
    print("Grade 16+")
else:
    print(f"Grade is {index}")